### PR TITLE
feat(text): Split results into sections by result type

### PR DESCRIPTION
**Pull request by underyx**
This is a nice representative snapshot for these changes:
https://github.com/returntocorp/semgrep/pull/7300/files#diff-58fb505c6987618a0e78416d0d2e5496279c095757f2be5111be0138a131eeb4

PR checklist:

- [ ] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
What is the goal of this PR? What is the test plan?

**Comment by underyx** [comment]
> What is the goal of this PR? What is the test plan?
> 
> 

Just that you see some changes early. This is all my work from my flight home. I'll split it into smaller prs.

File: cli/tests/e2e/snapshots/test_check/test_terminal_output_quiet/results.txt [hunk 1]
```diff
@@ -7,47 +7,9 @@ SEMGREP_SETTINGS_FILE="<MASKED>" SEMGREP_FORCE_COLOR="true" SEMGREP_USER_AGENT_A
 === end of exit code
 
 === stdout - plain
-Findings:
```
Comment (brendongo) [inline cli/tests/e2e/snapshots/test_check/test_terminal_output_quiet/results.txt:10]: Is this intended? Findings just went away

File: cli/tests/e2e/snapshots/test_ci/test_config_run/noautofix/results.txt [hunk 2]
```diff
@@ -7,57 +7,65 @@ SEMGREP_APP_TOKEN="" SEMGREP_USER_AGENT_APPEND="pytest" SEMGREP_SETTINGS_FILE="<
...
-         19┆ baz == 4
-          ⋮┆----------------------------------------
-     taint-test
-        unsafe use of danger
-
-         27┆ sink(d2)
+                                    
+                                    
+┌──────────────────────────────────┐
+│ 1 Reachable Supply Chain Finding │
```
Comment (brendongo) [inline cli/tests/e2e/snapshots/test_ci/test_config_run/noautofix/results.txt]: This is an unreachable finding

File: cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/results.txt [hunk 3]
```diff
@@ -7,53 +7,61 @@ BUILD_BUILDID="some_id" SEMGREP_REPO_NAME="a/repo/name" SEMGREP_REPO_URL="https:
...
+    foo.py 
+       eqeq-five                 
+          useless comparison to 5
+                                 
+           ▶▶┆ Autofix ▶ (x == 2)
+           15┆ (x == 2)
+                            
+                            
+┌──────────────────────────┐
+│ 6 Blocking Code Findings │
```
Comment (brendongo) [inline cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/results.txt:13]: Have we considered language that would better imply that the rule is blocking not the finding per se. Like "6 Code Findings from Blocking Rules" or is that too verbose?

File: cli/tests/e2e/snapshots/test_shouldafound/test_shouldafound_findings_output/None-X__X/results.txt [hunk 4]
```diff
@@ -7,14 +7,19 @@ SEMGREP_SETTINGS_FILE="<MASKED>" SEMGREP_USER_AGENT_APPEND="pytest" SEMGREP_VERS
 === end of exit code
 
 === stdout - plain
-Findings:
+                   
+                   
+┌─────────────────┐
+│ 3 Code Findings │
```
Comment (brendongo) [inline cli/tests/e2e/snapshots/test_shouldafound/test_shouldafound_findings_output/None-X__X/results.txt:13]: Why does this one just say Code Findings no blocking or non blocking?

Comment (underyx) [reply cli/tests/e2e/snapshots/test_shouldafound/test_shouldafound_findings_output/None-X__X/results.txt:13]: `semgrep scan` has no concept of blocking/non-blocking, only `semgrep ci` does that

File: cli/tests/e2e/snapshots/test_check/test_terminal_output_quiet/results.txt (hunk 1 above)
Comment (underyx) [reply cli/tests/e2e/snapshots/test_check/test_terminal_output_quiet/results.txt:10]: oof cause the console respects `--quiet`, right

File: cli/tests/e2e/snapshots/test_ci/test_config_run/noautofix/results.txt (hunk 2 above)
Comment (underyx) [reply cli/tests/e2e/snapshots/test_ci/test_config_run/noautofix/results.txt]: The rule is `sca-kind: upgrade-only.` so this should be reported as reachable actually. I agree we should add some clear indication of it being always reachable, but not sure if it has to be in this PR.

File: cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/results.txt (hunk 3 above)
Comment (underyx) [reply cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/results.txt:13]: I think that is too verbose, yes.

File: cli/tests/e2e/snapshots/test_check/test_terminal_output_quiet/results.txt (hunk 1 above)
Comment (underyx) [reply cli/tests/e2e/snapshots/test_check/test_terminal_output_quiet/results.txt:10]: fixed!

**Review by brendongo — APPROVED** [review]

**Pull request merged**